### PR TITLE
deps: dedup object, wasmparser, wasm-encoder

### PR DESCRIPTION
**Pull request by klensy**
* dedups one `object`, additional dupe will be removed, with next `thorin-dwp` update
* `wasmparser` pinned to minor versions, so full merge isn't possible
* same with `wasm-encoder`

Turned off some features for `wasmparser` (see features https://github.com/bytecodealliance/wasm-tools/blob/v1.208.1/crates/wasmparser/Cargo.toml) in `run-make-support`, looks working?

### CODE REVIEW

**Comment by klensy** [comment]
`libpanic_unwind-4f01f359c61a0a5e.rlib(lib.rmeta): SHT_STRTAB string table section [index 4] is empty` Oh.

**Comment by klensy** [comment]
Sorry for ping, wrong rebase.

**Comment by klensy** [comment]
@rustbot ready

there was an issue with `object`, that was fixed by https://github.com/gimli-rs/object/pull/710

File: compiler/rustc_codegen_ssa/src/back/archive.rs [hunk 1]
```diff
@@ -146,17 +144,23 @@ pub fn try_extract_macho_fat_archive(
         _ => return Ok(None),
     };
 
-    match object::macho::FatHeader::parse(&*archive_map) {
-        Ok(h) if h.magic.get(object::endian::BigEndian) == object::macho::FAT_MAGIC => {
-            let archs = object::macho::FatHeader::parse_arch32(&*archive_map);
+    if let Ok(h) = object::read::macho::MachOFatFile32::parse(&*archive_map) {
+        if h.header().magic.get(object::endian::BigEndian) == object::macho::FAT_MAGIC {
```
Comment (philipc) [inline compiler/rustc_codegen_ssa/src/back/archive.rs:148]: This check is already done at https://github.com/gimli-rs/object/blob/12d0948a288e38f2458bc5d99f8fea2405224424/src/read/macho/fat.rs#L37

Comment (klensy) [reply compiler/rustc_codegen_ssa/src/back/archive.rs]: https://github.com/rust-lang/rust/pull/127860/commits/616899dcc5abe89a9dd0fc3e3b00e1b2d95e995c

**Comment by Mark-Simulacrum** [comment]
r=me with some squashing/cleanup

**Pull request merged**